velox/expression/tests/ExprOptimizerTest.cpp

```diff
@@ -0,0 +1,497 @@
+  testExpression(
+      "array_sort(c0, x -> length(x) + (3 * 2))",
+      "array_sort(c0, x -> length(x) + 6)",
+      type,
+      lambdaTypes,
+      expectedLambdaTypes);
+  testExpression(
+      "array_sort(c0, x -> length(x) * abs(-1))",
+      "array_sort(c0, x -> length(x) * 1)",
+      type,
+      lambdaTypes,
+      expectedLambdaTypes);
+
+  lambdaTypes = {VARCHAR(), VARCHAR()};
+  testExpression(
+      "array_sort(c0, (x, y) -> if(length(x) < length(y), (2 - 3), if(length(x) > length(y), (-2 + 3), 0 * 3)))",
+      "array_sort(c0, x -> length(x))",
+      type,
+      lambdaTypes,
+      expectedLambdaTypes);
+  testExpression(
+      "array_sort(c0, (x, y) -> if(length(x) < length(y), (-1 * 1), if(length(x) = length(y), abs(0), (2 / 2))))",
+      "array_sort(c0, x -> length(x))",
+      type,
+      lambdaTypes,
+      expectedLambdaTypes);
+  testExpression(
+      "array_sort(c0, (x, y) -> if(length(x) < length(y), abs(-1), if(length(x) > length(y), -4 / 3, 0 / 3)))",
+      "array_sort_desc(c0, x -> length(x))",
+      type,
+      lambdaTypes,
+      expectedLambdaTypes);
+
+  type = ROW({"c0"}, {ARRAY(DOUBLE())});
+  lambdaTypes = {DOUBLE(), DOUBLE()};
+  testExpression(
+      "reduce(c0, (89.0E0 + 11.0E0), (s, x) -> s + x * (0.2E0 - 0.1E0), s -> (s < (100.0E0 + 1.0E0)))",
+      "reduce(c0, 100.0E0, (s, x) -> s + x * 0.1E0, s -> (s < 101.0E0))",
+      type,
+      lambdaTypes,
+      {DOUBLE(), DOUBLE()});
+  testExpression(
+      "reduce(c0, abs(-8.0E0), (s, x) -> (s + (4.0E0 / 2.0E0)) - x, s -> s)",
+      "8.0E0 + cast(array_sum_propagate_element_null(transform(c0, x -> 2.0E0 - x)) AS DOUBLE)",
+      type,
+      lambdaTypes,
+      {DOUBLE()});
+
+  type = ROW({"c0"}, {ARRAY(INTEGER())});
+  lambdaTypes = {INTEGER(), INTEGER()};
+  testExpression(
+      "reduce(c0, 100 / 10, (s, x) -> s + x, s -> s + (3 - 2))",
+      "reduce(c0, 10, (s, x) -> s + x, s -> s + 1)",
+      type,
+      lambdaTypes,
+      lambdaTypes);
+  testExpression(
+      "reduce(c0, 8 / 2, (s, x) -> if(x % 2 = 0, s + 1, s), s -> s)",
+      "4 + cast(array_sum_propagate_element_null(transform(c0, x -> if(x % 2 = 0, 1, 0))) AS BIGINT)",
+      type,
+      lambdaTypes,
+      {INTEGER()});
+
+  lambdaTypes = {SMALLINT(), SMALLINT()};
+  testExpression(
+      "reduce(c0, 1 - 1, (s, x) -> s + x, s -> coalesce(s, abs(2 - 10), 5 * 3) * (5 * 2))",
+      "reduce(c0, 0, (s, x) -> s + x, s -> coalesce(s, 8, 15) * 10)",
+      ROW({"c0"}, {ARRAY(SMALLINT())}),
+      lambdaTypes,
+      lambdaTypes);
+  testExpression(
+      "reduce(c0, 15 * 3, (s, x) -> s + x * abs(-2), s -> s)",
+      "45 + cast(array_sum_propagate_element_null(transform(c0, x -> x * 2)) AS BIGINT)",
+      ROW({"c0"}, {ARRAY(TINYINT())}),
+      {TINYINT(), TINYINT()},
+      {TINYINT()});
+}
+
+/// Test to ensure session queryCtx is used during expression optimization.
+TEST_F(ExprOptimizerTest, queryCtx) {
+  setQueryTimeZone("Pacific/Apia");
```

Comment:
Add a lambda function with the input textExpressions that passes the timezone and result values as parameter.